### PR TITLE
 salt: 2017.7.2 -> 2017.7.4, fix msgpack dependency

### PR DESCRIPTION
**Pull request by aneeshusa**
###### Motivation for this change

Fix Salt build. Bump Salt to the latest release.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

It looks like this package was missed in #34809; the latest (2017.7.3) still seems to require `msgpack-python`.
cc @danbst and @PierreR as Salt users: I won't be able to test this thoroughly for a bit, could you give this branch a spin and report how it goes?

### CODE REVIEW

File: pkgs/tools/admin/salt/default.nix [hunk 1]
```diff
@@ -8,19 +8,19 @@
 
 python2Packages.buildPythonApplication rec {
   pname = "salt";
-  version = "2017.7.2";
+  version = "2017.7.3";
   name = "${pname}-${version}";
```
Comment (dotlambda) [inline pkgs/tools/admin/salt/default.nix]: Please remove `name`, it is not needed for `buildPython*`

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
There is a 2017.7.4 version: https://pypi.org/project/salt/

**Comment by aneeshusa** [comment]
Thanks for the review @dotlambda, updated the PR. (I did build locally but still won't be able to test for a while.)

**Comment by dotlambda** [comment]
@aneeshusa Did you test yet?

**Pull request merged**